### PR TITLE
W 12689960 New field at messages: MessageId 

### PR DESCRIPTION
**Pull request by damianpedra**
New field at messages: MessageId

### CODE REVIEW

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async2MessageParser.scala [hunk 1]
```diff
@@ -298,6 +315,31 @@ abstract class Async21MessagePopulator()(implicit ctx: AsyncWebApiContext) exten
...
+    map.key("messageId").foreach { entry =>
+      val messageId = entry.value.toString()
+          if (!ctx.registerMessageId(messageId))
+            ctx.eh.violation(
+              ParserSideValidations.DuplicatedMessageId, message, s"duplicado $messageId", entry.value.location
+            )
+      parseMessageId(map, message)
+    }
+    message
+//    map.key("messageId", MessageModel.MessageId in message)
```
Comment (looseale) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async2MessageParser.scala]: Remove commented lines

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async2MessageParser.scala [hunk 2]
```diff
@@ -298,6 +315,31 @@ abstract class Async21MessagePopulator()(implicit ctx: AsyncWebApiContext) exten
...
+abstract class Async24MessagePopulator()(implicit ctx: AsyncWebApiContext) extends Async21MessagePopulator {
+  override protected val closeShapeName: String = "message24"
+
+  override def populate(map: YMap, message: Message): Message = {
+    super.populate(map, message)
+    map.key("messageId").foreach { entry =>
+      val messageId = entry.value.toString()
+          if (!ctx.registerMessageId(messageId))
+            ctx.eh.violation(
+              ParserSideValidations.DuplicatedMessageId, message, s"duplicado $messageId", entry.value.location
```
Comment (looseale) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async2MessageParser.scala]: Fix error message "duplicado"

File: amf-cli/shared/src/test/scala/amf/validation/Async20UniquePlatformUnitValidationsTest.scala [hunk 3]
```diff
@@ -297,44 +297,20 @@ class Async20UniquePlatformUnitValidationsTest extends UniquePlatformReportGenTe
...
 
-  test("Valid ref to operation with another ref") {
-    validate("double-references/valid-ref-to-operation-with-ref.yaml")
-  }
-
-  test("Valid ref to parameter with another ref") {
-    validate("double-references/valid-ref-to-parameter-with-ref.yaml")
-  }
-
-  test("Valid ref to bindings with another ref") {
```
Comment (looseale) [inline amf-cli/shared/src/test/scala/amf/validation/Async20UniquePlatformUnitValidationsTest.scala]: Why all these deleted lines?

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async2MessageParser.scala (hunk 1 above)
Comment (damianpedra) [reply amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async2MessageParser.scala]: fixex

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async2MessageParser.scala (hunk 2 above)
Comment (damianpedra) [reply amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async2MessageParser.scala]: fixed

File: amf-cli/shared/src/test/scala/amf/validation/Async20UniquePlatformUnitValidationsTest.scala (hunk 3 above)
Comment (damianpedra) [reply amf-cli/shared/src/test/scala/amf/validation/Async20UniquePlatformUnitValidationsTest.scala]: fixed

**Review by looseale — APPROVED** [review]

**Pull request merged**